engineioxide/src/service.rs

```diff
@@ -222,8 +229,32 @@ impl RequestInfo {
     /// Parse the request URI to extract the [`TransportType`](crate::service::TransportType) and the socket id.
     fn parse<B>(req: &Request<B>) -> Result<Self, Error> {
         let query = req.uri().query().ok_or(UnknownTransport)?;
-        if !query.contains("EIO=4") {
-            return Err(UnsupportedProtocolVersion);
+
+        let protocol: ProtocolVersion = query
```

Comment:
It needs some unit tests for all the `ProtocolVersion` cases like `TransportType` (see below)